ethereum/slogging.py

```diff
@@ -138,11 +85,228 @@ def pop_records(self):
+    name_levels = [('', logging.getLevelName(rootLogger.level))]
+
+    for name, logger in list(rootLogger.manager.loggerDict.items()):
+        name_levels.append((name, logging.getLevelName(logger.level)))
+    config_string = ','.join('%s:%s' % x for x in name_levels)
+    #struct_root = get_logger().bind()
+    #log_json = bool([p for p in struct_root._processors if
+    #                 isinstance(p, structlog.processors.JSONRenderer)])
+    return dict(config_string=config_string, log_json=rootLogger.log_json)
+
+# helpers
+known_loggers = set()  # know to structlog (i.e. maybe not yet initialized w/ logging)
+
+
+def get_logger_names():
+    # logging.Logger.manager.loggerDict.keys() # used ones
+    # initialized at module load get_logger
+    return sorted(known_loggers, key=lambda x: '' if not x else x)
+
+#----------------------------- from standart logging module width some modifications ----------------------------
+class ethLogger(logging.Logger):
+    def __init__(self, name, level=DEFAULT_LOGLEVEL):
+        super(ethLogger, self).__init__(name, level=level)
+
+    def is_active(self, level_name='trace'):
+        return self.isEnabledFor(logging._checkLevel(level_name.upper()))
+        #getLogger()
+        #self.setLevel(getattr(logging, level_name.upper()))
 
 
+class RootLogger(ethLogger):
+    """
+    A root logger is not that different to any other logger, except that
+    it must have a logging level and there is only one instance of it in
+    the hierarchy.
+    """
+    def __init__(self, level, log_json=False):
+        """
+        Initialize the logger with the name "root".
+        """
+        ethLogger.__init__(self, "root", level)
+        self.log_json = False
+
+class ethManager(logging.Manager):
+    def __init__(self, rootnode):
+        self.loggerClass = ethLogger
+        super(ethManager, self).__init__(rootnode)
+
+    def getLogger(self, name):
+        logging.setLoggerClass(ethLogger)
+        return super(ethManager, self).getLogger(name)
+
+rootLogger = RootLogger(DEFAULT_LOGLEVEL)
+ethLogger.root = rootLogger
+ethLogger.manager = ethManager(ethLogger.root)
+
+def help_make_kws(kws, name, msg, log_json=False):
+    new_kws = {}
+    message = ""
+
+    if log_json == True:
+        message = {}
+        for i in kws:
+            if i not in formatter_list:  # levelno, asctime......
+                print i, "=", kws[i]
+                #s = " {i}={kwsi}".format(i=i, kwsi=str(kws[i]))
+                #message += s
+                #message[i] = repr(kws[i]) if type(kws[i]).__name__=='instance'   else kws[i]
+                #if type(kws[i]) == (types.IntType or types.LongType or types.FloatType or types.ComplexType or types.DictionaryType):
+                #    message[i] = kws[i] # for json dumps
+                #else:
+                #    message[i] = repr(kws[i]) # for json dumps
+                kws_i_d = getattr(kws[i], "__dict__", False)
+                if kws_i_d != False:
```

Comment:
`None` would probably be a better default value than `False`. At least change the condition to use `is not` instead of `!=`.